contracts/tokenfactory-issuer/src/execute.rs

```diff
@@ -21,20 +21,26 @@ pub fn mint(
 
     // don't allow minting of 0 coins
     if amount.is_zero() {
-        return Result::Err(ContractError::ZeroAmount {});
+        return Err(ContractError::ZeroAmount {});
     }
 
     // decrease minter allowance
     // if minter allowance goes negative, throw error
-    MINTER_ALLOWANCES.update(deps.storage, &info.sender, |allowance| {
-        allowance
-            .unwrap_or_else(Uint128::zero)
-            .checked_sub(amount)
-            .map_err(StdError::overflow)
-    })?;
+    // if minter allowance goes 0, remove from storage
+    let allowance = MINTER_ALLOWANCES.may_load(deps.storage, &info.sender)?;
+    let updated_allowance = allowance
+        .unwrap_or_else(Uint128::zero)
+        .checked_sub(amount)
+        .map_err(StdError::overflow)?;
```

Comment:
We should probably make an explicit contract error type for this

Reply:
That make sense, maybe `NotEnoughAllowance`

Reply:
@sunnya97 updated, does that looks good?